server/sonar-server/src/main/java/org/sonar/server/platform/Platform.java

```diff
@@ -157,10 +157,12 @@ public Status status() {
     if (!started) {
       return Status.BOOTING;
     }
-    if (levelSafeMode != null && currentLevel == levelSafeMode) {
+    PlatformLevel current = this.currentLevel;
+    PlatformLevel levelSafe = this.levelSafeMode;
+    if (levelSafe != null && current == levelSafe) {
       return Status.SAFEMODE;
     }
-    if (currentLevel == level4) {
+    if (current == level4) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)